### PR TITLE
#169258584 Super admin should be able to assign roles

### PR DESCRIPTION
**Pull request by victkarangwa**
#### What does this PR do?
- Super admin can asign new roles to the users
#### Description of Task to be completed?
- This PR adds the functionalities for the super admin to assign new roles to the staff members
#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-frontend`
- `git checkout ft-user-role-settings-169258584`
- `npm run install` to install all dependencies
- `npm run dev` to start development server
- `login with super admin credentials
- navigate to `users` or `/admin/roles` route

NOTE: You won't be able to access the page unless you login in as a super admin
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#169258584](https://www.pivotaltracker.com/story/show/169258584)
#### Screenshots (if appropriate)
Screen Shot 2020-01-23 at 21 52 43
Screen Shot 2020-01-23 at 21 52 59

#### Questions:
N/A

### CODE REVIEW

**Review by NiyongaboEric — CHANGES_REQUESTED** [review]
Well done @victorkarangwa4. I would like to request you to improve work again.

- Please ensure we have pagination functionality, for now, we are displaying all users from DB. As an admin, I would like to see a few users, then I paginate to get more users.

- What do you think if we toast user who are trying to access admin page? What you have done is to protect the route but it is not easy to guess why the user is being redirected to the dashboard.

- I have noticed that we can use blocked token to access admin route.

**Review by nziokaivy — CHANGES_REQUESTED** [review]
Great job @Victor! Very nice implementation!

However please ensure we can view the users' role

**Review by NiyongaboEric — APPROVED** [review]

File: src/__tests__/components/SelectOption.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import SelectOption from '../../components/SelectOption';
+
+describe('Test suite for Select option component', () => {
+  const options = [
+    { id: 6, type: 'manager' },
+    { id: 3, type: 'user' },
+  ];
```
Comment (hezronkimutai) [inline src/__tests__/components/SelectOption.test.js]: You can consider moving this data to the __mocks__ folder

File: src/__tests__/containers/admin/UserRoles.test.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    const componentDidMountSpy = jest.spyOn(component.instance(), 'componentDidMount');
+    component.find('[data-test="userRoles-test"]');
+    expect(componentDidMountSpy).not.toHaveBeenCalled();
+  });
+  // it('Should call componentDidUpdate Successfully', () => {
+  //   const component = setUp({});
+  //   component.setProps(updateDefaultProps);
+  //   const componentDidUpdateSpy = jest.spyOn(component.instance(), 'componentDidUpdate');
+  //   expect(componentDidUpdateSpy).not.toHaveBeenCalled();
+  // });
```
Comment (hezronkimutai) [inline src/__tests__/containers/admin/UserRoles.test.js]: Consider removing these comments too

File: src/views/trips/CommonRequests.js [hunk 3]
```diff
@@ -3,7 +3,18 @@ import { Link } from 'react-router-dom';
...
+      data,
+      navs,
+      tableHeads,
+      entities,
+      paginationEnd,
+      paginationStart,
+      paginatedRequest,
+      paginationAction,
+    } = this.props;
+    console.log('/////', entities);
```
Comment (hezronkimutai) [inline src/views/trips/CommonRequests.js]: Remove these console log too.

**Review by hezronkimutai — CHANGES_REQUESTED** [review]
Hi @ victor I left some comments please have a look at it.

**Review by nziokaivy — APPROVED** [review]

**Review by hezronkimutai — APPROVED** [review]

**Review by shemaeric — APPROVED** [review]
Good work @victorkarangwa4

File: src/components/SelectOption.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      </option>
+      {options
+      && options.map((item) => (
+        <option value={item.id} key={item.id}>
+          {item.type}
+        </option>
+      ))}
+    </select>
+  );
+};
```
Comment (william000000) [inline src/components/SelectOption.js]: Nice shot!

**Review by william000000 — APPROVED** [review]

File: src/components/SelectOption.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      <option disabled value="">
+      Choose
+      </option>
+      {options
+      && options.map((item) => (
+        <option value={item.id} key={item.id}>
+          {item.type}
+        </option>
+      ))}
+    </select>
```
Comment (william000000) [inline src/components/SelectOption.js:18]: Nice Shot!

**Pull request merged**